libs/pdi-scanner/src/client.rs

```diff
@@ -0,0 +1,921 @@
+            timeout
+        )
+    }
+
+    /// Gets the double sheet detection calibration value for a single sheet of paper. This value
+    /// should be lower than the value for two sheets of paper.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the response is not received within the timeout.
+    pub fn get_double_feed_detection_single_sheet_calibration_value(
+        &mut self,
+        timeout: Duration,
+    ) -> Result<u16> {
+        send_and_recv!(
+            self => Outgoing::GetDoubleFeedDetectionSingleSheetCalibrationValueRequest,
+            Incoming::GetDoubleFeedDetectionSingleSheetCalibrationValueResponse(value) => value,
+            timeout
+        )
+    }
+
+    /// Gets the double sheet detection calibration value for two sheets of
+    /// paper. This value should be higher than the value for a single sheet of
+    /// paper.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the response is not received
+    /// within the timeout.
+    pub fn get_double_feed_detection_double_sheet_calibration_value(
+        &mut self,
+        timeout: Duration,
+    ) -> Result<u16> {
+        send_and_recv!(
+            self => Outgoing::GetDoubleFeedDetectionDoubleSheetCalibrationValueRequest,
+            Incoming::GetDoubleFeedDetectionDoubleSheetCalibrationValueResponse(value) => value,
+            timeout
+        )
+    }
+
+    /// Gets the double sheet detection threshold value. Values above this
+    /// threshold are considered to be double feeds.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the response is not received
+    /// within the timeout.
+    pub fn get_double_feed_detection_double_sheet_threshold_value(
+        &mut self,
+        timeout: Duration,
+    ) -> Result<u16> {
+        send_and_recv!(
+            self => Outgoing::GetDoubleFeedDetectionDoubleSheetThresholdValueRequest,
+            Incoming::GetDoubleFeedDetectionDoubleSheetThresholdValueResponse(value) => value,
+            timeout
+        )
+    }
+
+    /// Enables or disables the array light source.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the connection to the scanner is
+    /// lost.
+    pub fn set_array_light_source_enabled(&mut self, enabled: bool) -> Result<()> {
+        self.send(if enabled {
+            Outgoing::TurnArrayLightSourceOnRequest
+        } else {
+            Outgoing::TurnArrayLightSourceOffRequest
+        })
+    }
+
+    /// Connects to the scanner using the same commands that were captured by
+    /// wireshark from the `scan_demo` program. Some of them may not be
+    /// necessary, but they're included for now.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if any of the commands fail to
+    /// validate or if the response is not received within the timeout.
+    pub fn send_connect(&mut self) -> Result<()> {
```

Comment:
What do you think we should do with `send_connect` and `send_enable_scan_commands`? Should we keep using them or try going without and adding in the bare minimum necessary?

(accidentally missed this file in my first review because it was collapsed)

Reply:
I went with this approach initially because it minimized the sources of issues that could arise when building our own driver. To the extent we think we understand the interaction model I think we can move to customize this a bit. For now I'm inclined to leave it, but I don't feel strongly about it.

Reply:
Leaving it for now sounds good to me. I wonder if this is an area where PDI could provide some insight so we're not guessing as much?

Reply:
Probably so, yeah.